crates/prover/src/components/store_deref_fp.rs

```diff
@@ -0,0 +1,468 @@
+use cairo_m_common::Opcode;
+use num_traits::{One, Zero};
+use rayon::iter::{
+    IndexedParallelIterator, IntoParallelIterator, IntoParallelRefIterator, ParallelIterator,
+};
+use serde::{Deserialize, Serialize};
+use stwo_air_utils::trace::component_trace::ComponentTrace;
+use stwo_air_utils_derive::{IterMut, ParIterMut, Uninitialized};
+use stwo_prover::constraint_framework::logup::LogupTraceGenerator;
+use stwo_prover::constraint_framework::{
+    EvalAtRow, FrameworkComponent, FrameworkEval, Relation, RelationEntry,
+};
+use stwo_prover::core::backend::simd::conversion::Pack;
+use stwo_prover::core::backend::simd::m31::{PackedM31, LOG_N_LANES, N_LANES};
+use stwo_prover::core::backend::simd::qm31::PackedQM31;
+use stwo_prover::core::backend::simd::SimdBackend;
+use stwo_prover::core::backend::BackendForChannel;
+use stwo_prover::core::channel::{Channel, MerkleChannel};
+use stwo_prover::core::fields::m31::{BaseField, M31};
+use stwo_prover::core::fields::qm31::SecureField;
+use stwo_prover::core::fields::secure_column::SECURE_EXTENSION_DEGREE;
+use stwo_prover::core::pcs::TreeVec;
+use stwo_prover::core::poly::circle::CircleEvaluation;
+use stwo_prover::core::poly::BitReversedOrder;
+
+use crate::adapter::StateData;
+use crate::relations;
+use crate::utils::PackedStateData;
+
+const N_TRACE_COLUMNS: usize = 13;
+// -(pc, [opcode_id, off0, off1, off2], prev_clock) || +(pc, [opcode_id, off0, off1, off2], clock)
+// -(fp+off2, [prev_value, 0, 0, 0], prev_clock) || +(fp+off2, [value, 0, 0, 0], clock)
+// -(fp+off0, [value, 0, 0, 0], prev_clock) || +(fp+off0, [value, 0, 0, 0], clock)
+const N_MEMORY_LOOKUPS: usize = 2 * 3;
+const N_REGISTERS_LOOKUPS: usize = 2; // -(pc, fp) || +(pc+1, fp)
+
+const LOOKUPS_COLUMNS: usize = N_MEMORY_LOOKUPS + N_REGISTERS_LOOKUPS;
+
+pub struct InteractionClaimData {
+    pub lookup_data: LookupData,
+    pub non_padded_length: usize,
+}
+
+#[derive(Uninitialized, IterMut, ParIterMut)]
+pub struct LookupData {
+    // 6 elements: addr, value_0, value_1, value_2, value_3, clock
+    pub memory: [Vec<[PackedM31; 6]>; N_MEMORY_LOOKUPS],
```

Comment:
as per @ClementWalter 's refactor, change the order everywhere to addr, clock, ...values